test/kino_db/connection_cell_test.exs

```diff
@@ -60,6 +60,35 @@ defmodule KinoDB.ConnectionCellTest do
                {:ok, db} = Kino.start_child({Exqlite, opts})\
                """
     end
+
+    test "restores source code from attrs with BigQuery" do
+      attrs = %{
+        "variable" => "db",
+        "type" => "bigquery",
+        "project_id" => "",
+        "credentials" => %{},
+        "default_dataset_id" => ""
+      }
+
+      {_kino, source} = start_smart_cell!(ConnectionCell, attrs)
+
+      assert source ==
+               """
+               credentials = %{}
+
+               opts = [
+                 name: ReqBigQuery.Goth,
+                 http_client: &Req.request/1,
+                 source: {:service_account, credentials, []}
+               ]
+
+               db =
+                 Req.new(http_errors: :raise)
+                 |> ReqBigQuery.attach(goth: ReqBigQuery.Goth, project_id: "", default_dataset_id: "")
+
+               {:ok, _goth_pid} = Kino.start_child({Goth, opts})\
```

Comment:
Good question, one one hand I can imagine it's fairly large, on the other it would actually make sense that the connection cell outputs whatever the connection is, while goth pid is not really relevant :thinking:

Reply:
The output is huge and show all information from given `options`, but doesn't show the credentials (🙏🏻)

Reply:
If we don't like the large output we can also do:

```elixir
{:ok, _pid} = Kino.start_child({Goth, opts})

db = ...

:ok
```

I'm not totally sure, @wojtekmach, @josevalim any preference?

Reply:
@wojtekmach should we implement the protocol for `Req.Request` to show only some data?

Reply:
+1 for moving code around, I think goth_opts and starting goth should be together.

I think showing Req.Request is interesting but yeah, the struct is quite large. I'm not sure how to decide what to show and what not to.

For lack of other options yeah, I'd just return `:ok` and move on, maybe as we add more smart cells we'll figure out something better.